src/components/MyAccount/EditPopUpWindow.tsx

```diff
@@ -0,0 +1,218 @@
+import { useStoreSelector } from "@/store/hooks";
+import { updateUserInfo, userData } from "@/store/reducer/userSlice";
+import {
+  Button,
+  Modal,
+  ModalBody,
+  ModalCloseButton,
+  ModalContent,
+  ModalFooter,
+  ModalHeader,
+  ModalOverlay,
+  useToast,
+} from "@chakra-ui/react";
+import { AxiosResponse } from "axios";
+import React, { useState } from "react";
+import { useDispatch } from "react-redux";
+import { updateUser, validPassword } from "../Login/helpers/userRequests";
+import validatePwd from "../Login/helpers/validatePwd";
+import EditInputGroup from "./EditInputGroup";
+
+const EditPopUpWindow = ({
+  onClose,
+  isOpen,
+  title,
+}: {
+  onClose: () => void;
+  isOpen: boolean;
+  title: string;
+}) => {
+  const [newFirstName, setNewFirstName] = useState("");
+  const [newLastName, setNewLastName] = useState("");
+  const [oldPsw, setOldPsw] = useState("");
+  const [newPsw, setNewPsw] = useState("");
+  const [confirmPsw, setConfirmPsw] = useState("");
+  const [errorMessage, setErrorMessage] = useState("");
+  const [weakPasswordMsg, setWeakPasswordMsg] = useState("");
+  const [incorrectPswMsg, setIncorrectPswMsg] = useState("");
+  const currentUserId = useStoreSelector(userData).userId;
+  const toast = useToast();
+  const dispatch = useDispatch();
+
+  const isBlank = (): boolean => {
+    switch (title) {
```

Comment:
Don't use  Mutable "title" for switch case judge. You should pass "type" into this component. Because "title" will change if i18n change or other reasons.

Reply:
It will not affect the switch case judge if title is changed. We can change title into any value.

Reply:
I don't think so. Perhaps other people change "title" value but forget to change this switch cases. After all, it is not Robustness.

Reply:
Fix that, please check